pkg/gardenlet/controller/shootstate/add.go

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shootstate
+
+import (
+	"fmt"
+
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	"github.com/gardener/gardener/pkg/gardenlet/apis/config"
+	shootsecret "github.com/gardener/gardener/pkg/gardenlet/controller/shootstate/secret"
+)
+
+// AddToManager adds Controllers to the given manager.
+func AddToManager(
+	mgr manager.Manager,
+	gardenCluster cluster.Cluster,
+	seedCluster cluster.Cluster,
+	cfg config.GardenletConfiguration,
+) error {
+	if err := (&shootsecret.Reconciler{
+		Config: *cfg.Controllers.ShootSecret,
+	}).AddToManager(mgr, gardenCluster, seedCluster); err != nil {
+		return fmt.Errorf("failed adding ShootSecret controller: %w", err)
```

Comment:
```suggestion
		return fmt.Errorf("failed adding secret reconciler: %w", err)
```

Reply:
Should I change the package and controller name also to `secret` instead of `shootsecret`?

Reply:
Yes, the `package` should be `secret` and the controller name should be `shootstate-secret`